Intermittent failure of TestListen/Serve_calls_s.listen_if_necessary

If we never see this in CI, I don't care, but if we do, we should fix or skip this test:
```
$ while go test -count=1 ./...; do true; done;
?   	github.com/packethost/pkg/cmd/examplelog	[no test files]
ok  	github.com/packethost/pkg/env	0.005s
--- FAIL: TestListen (0.00s)
    --- FAIL: TestListen/Serve_calls_s.listen_if_necessary (0.00s)
        grpc_test.go:360:
            	Error Trace:	grpc_test.go:360
            	Error:      	Received unexpected error:
            	            	rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:8080: connect: connection refused"
            	Test:       	TestListen/Serve_calls_s.listen_if_necessary
FAIL
FAIL	github.com/packethost/pkg/grpc	0.243s
ok  	github.com/packethost/pkg/internal/testenv	0.005s
ok  	github.com/packethost/pkg/log	0.015s
ok  	github.com/packethost/pkg/log/internal/rollbar	0.004s
ok  	github.com/packethost/pkg/prometheus/promdb	0.005s
FAIL
```

--- a/grpc/grpc_test.go
+++ b/grpc/grpc_test.go
@@ -347,6 +347,7 @@ func TestListen(t *testing.T) {
 		s, err := NewServer(l, defSrv)
 		assert.NoError(t, err)
 		assert.NotNil(t, s)
+		s.port = 0
 
 		go func() {
 			if err := s.Serve(); err != nil {

